chore: 添加bazel构建系统

lazychih114 opened the pull request.

lazychih114 committed:
chore:添加bazel构建系统

lazychih114 committed:
chore:添加bazel构建系统

lazychih114 committed:
Merge branch 'notURandomDev:main' into main

lazychih114 committed:
Merge branch 'notURandomDev:main' into main

lazychih114 committed:
chore: 添加bazel构建系统

notURandomDev commented on RustSDK/rust_sdk.h:
这份文件位于RustSDK/include/rust_sdk.h中，不用重新写一份

notURandomDev commented on LarkClone/Frameworks/Core/LarkSDK/RustBridge.swift:
```diff
@@ -8,6 +8,8 @@
 import Foundation
 import LarkSDKPB
 import LarkBridgeModels
+import LarkBridge
```
项目中没有LarkBridge框架，这里的引用是什么意思？

lazychih114 replied:
暴露的Module库，然后显示引用

notURandomDev commented on LarkClone/Frameworks/Core/LarkSDK/module.modulemap:
为什么要将LarkSDK框架引用的名字改成LarkBridge？项目中还有很多其它的文件的引用都是通过 LarkSDK 进行 import 的，在这里进行改动的话，会使其它框架失去对于 LarkSDK 的引用

lazychih114 replied:
不会 我没改 只是构建的时候 我要额外处理桥接模块

notURandomDev commented on build.sh:
```diff
@@ -23,9 +25,23 @@ clean_build() {
+    ./script/setup_rust.sh
+}
+
+generate(){
```
mock_data数据脚本的运行，应该在MessengerTab和MailTab的构建过程中分别运行，而不是在LarkClone主应用的构建中运行

notURandomDev commented on build.sh:
```diff
@@ -23,9 +25,23 @@ clean_build() {
     bazel clean --expunge
 }
 
+setupenv(){
```
1. 环境的配置脚本，应该在 LarkSDK 的构建阶段运行，而不是主应用的构建阶段运行
2. 没有使用 `check_rust_built.sh` 脚本进行 RustSDK 构建的检查

lazychih114 replied:
首先这个脚本只是针对app的，如果构建特定目标 直接使用bazel命令即可

lazychih114 committed:
fix: 修复xcode、bazel兼容性问题

lazychih114 committed:
fix: 停止跟踪缓存文件

lazychih114 committed:
feat: 添加bazel环境安装

lazychih114 committed:
fix: 完善bazel构建脚本

lazychih114 committed:
fix: 取消缓存上传

lazychih114 committed:
fix: 修改gitignore文件

lazychih114 committed:
fix: 修复bazel环境重复配置问题

Commit:
feat: 为LarkColor添加BUILD.bazel

lazychih114 committed:
fix: 修复bazel版本依赖问题

lazychih114 committed:
fix: 修复bazel版本依赖问题

Commit:
fix：更新LarkAvatar的框架中文件的Target

Commit:
fix：将LarkColor的文件从LarkClone中移除

Commit:
refactor: 将ChatBubbleView提取为框架LarkChatBubble

- 在主应用的Compile Sources中删除了LarkChatBubble相关的文件
- 在引用LarkChatBubble的地方加入了import
- 声明调用LarkChatBubble框架的依赖

Commit:
refactor：将SearchBarView相关文件使用LarkSeachBar.framework进行组织

在主应用target的compile sources中删除了SearchBarView相关文件

lazychih114 committed:
Merge branch 'notURandomDev:main' into main

lazychih114 committed:
Merge branch 'refactor/frameworks' of https://github.com/notURandomDev/LarkClone

lazychih114 committed:
fix: bazel构建模块化

lazychih114 committed:
docs: 更新bazel构建使用文档

lazychih114 committed:
docs: 更新bazel构建使用文档

lazychih114 committed:
docs: 更新bazel构建使用文档

Commit:
fix：更新.gitignore中对于.bazel.bak文件的忽略

Commit:
feat：添加.bazelignore文件，对.DS_Store文件进行忽略

Commit:
fix：更新bazel规则中对于OS最低版本的要求

lazychih114 committed:
feature: 添加github action,生成app

lazychih114 committed:
fix: 修改bazel测试构建

Commit:
解决import在xcodebuild和bazel的二选一报错

通过条件宏，判断构建系统是bazel还是xcode，从而选择性地引入头文件

Commit:
fix：将新增的构建文件从Xcode目标中移除
